modules/dom.js

```diff
@@ -30,6 +30,58 @@ export const targetSelector = (element) => {
   }
 };
 
+function walk(element, checkFn) {
+  if (!element || element === document.body || element === document.documentElement) {
+    return undefined;
+  }
+  const checkValue = checkFn(element);
+  return checkValue || walk(element.parentElement, checkFn);
+}
+
+function isDialog(element) {
+  // doing it well
+  if (element.tagName === 'DIALOG') return true;
+  // making the best of it
+  if (element.getAttribute('role') === 'dialog') return true;
+  if (element.getAttribute('role') === 'alertdialog') return true;
+  if (element.getAttribute('aria-modal') === 'true') return true;
+  // doing it wrong
+  const computedStyle = window.getComputedStyle(element);
+  return (computedStyle && computedStyle.position === 'fixed' && computedStyle.zIndex > 100);
+}
+
+function isButton(element) {
+  if (element.tagName === 'BUTTON') return true;
+  if (element.tagName === 'INPUT' && element.getAttribute('type') === 'button') return true;
+  if (element.tagName === 'A') {
+    const classes = Array.from(element.classList);
+    return classes.some((className) => className.match(/button|cta/));
+  }
+  return element.getAttribute('role') === 'button';
+}
+
+function getSourceContext(element) {
+  if (element.closest('form')) return 'form';
+  if (element.closest('.block[data-block-name]')) return `.${element.closest('.block').getAttribute('data-block-name')}`;
```

Comment:
```
<div class=".block" block-name="ablock">
   ...
   <div class=".block">
      //element
```
would return null. Unlikely but could happen...

```suggestion
  const block = element.closest('.block[data-block-name]');
  if (block) return `.${block.getAttribute('data-block-name')}`;
```